cmd/sfncli/runner.go

```diff
@@ -78,6 +81,16 @@ func (t *TaskRunner) Process(ctx context.Context, args []string, input string) e
 	if executionName != nil {
 		t.execCmd.Env = append(os.Environ(), "_EXECUTION_NAME="+*executionName)
 	}
+	if t.workDirectory != "" {
+		// make a new tmpDir for every run
+		tmpDir, err := ioutil.TempDir(t.workDirectory, "")
+		if err != nil {
+			return err
+		}
+
+		t.execCmd.Env = append(t.execCmd.Env, "WORK_DIR="+t.workDirectory)
+		defer os.RemoveAll(tmpDir)
```

Comment:
is this safe to do for any user-provided workdirectory? should we maybe create a directory within the workdirectory to use as WORK_DIR so we can clean that up safely?

we can at least call out this assumption in the README so clients know that `workdirectory` is completely wiped after each run (i.e. don't use project root :P)

Reply:
It's a temp directory _in_ the provided directory. So even if you provide `/`, it should only be `/<some random number>`

```
» cat main.go
package main

import (
	"io/ioutil"
	"log"
)

func main() {
	dir, err := ioutil.TempDir("./", "")
	log.Printf("dir: '%s', err: '%s'", dir, err)
}
» go run main.go
2017/11/10 09:25:15 dir: '816108725', err: '%!s(<nil>)'
```

Reply:
ah perfect - i was reading that wrong!
should L91 be this instead then?:
```go
t.execCmd.Env = append(t.execCmd.Env, "WORK_DIR="+tmpDir)
```
instead of `"WORK_DIR="+t.workDirectory`?

Reply:
eek, yes.
Making a PR now